File: crates/nostr/examples/vanity.rs
// Copyright (c) 2022-2023 Yuki Kishimoto
// Distributed under the MIT software license

use nostr::prelude::*;

fn main() -> Result<()> {
    let num_cores = num_cpus::get();
    let keys = Keys::vanity(vec!["0000", "yuk", "yuk0"], true, num_cores)?;
    println!("Secret key: {}", keys.secret_key()?.to_bech32()?);
    println!("Public key: {}", keys.public_key().to_bech32()?);
    Ok(())
}

File: bindings/nostr-ffi/bindings-python/examples/vanity.py
from nostr import Keys

print("Mining keys...")
keys = Keys.vanity(["yuk0"], True, 8)

sk = keys.secret_key()
pk = keys.public_key()
print("Keys:")
print(" Public keys:")
print(f"     hex:    {pk.to_hex()}")
print(f"     bech32: {pk.to_bech32()}")
print(" Secret keys:")
print(f"     hex:    {sk.to_hex()}")
print(f"     bech32: {sk.to_bech32()}")
